crates/uv/src/commands/project/run.rs

```diff
@@ -1240,10 +1252,27 @@ impl RunCommand {
     }
 
     /// Convert a [`RunCommand`] into a [`Command`].
-    fn as_command(&self, interpreter: &Interpreter) -> Command {
+    fn as_command(&self, interpreter: &Interpreter, is_patch_request: bool) -> Command {
         match self {
             Self::Python(args) => {
-                let mut process = Command::new(interpreter.sys_executable());
+                let mut process = if !interpreter.is_standalone() || is_patch_request {
+                    Command::new(interpreter.sys_executable())
+                } else {
+                    let executable = interpreter
```

Comment:
Can you explain what's going on here? Why do we need this special handling in `run`?

Reply:
This is in order to support transparent upgrades for virtual envs created by `uv run python -m venv`. We need to run with the symlink path in order for it to be captured when the virtual environment is created.

Reply:
It seems critical to write a comment to that effect. It seems problematic that special-casing is needed here though, if that's necessary, won't our `~/.local/bin/python` executables never work?

Reply:
I'll need to explore it further, but it may be necessary to use shims in `~/.local/bin` in order to support transparently upgradeable virtual environments created with the `venv` module when calling them directly.

Reply:
I think we'll need a clear answer there before we can merge this implementation.

Reply:
That makes sense to me

Reply:
To be clear, I've done manual tests in the past using a shim, so I know that approach creates the upgradeable `.venv` structure (i.e., the correct `home` key and the correct symlinks in `.venv/bin`). For Windows, we'd be using the trampoline approach instead.

Reply:
I was able to do some further testing on this today. There appears to be no need for a shim and things should work with both unix installations and Windows trampoline-based installations into the `bin` directory.

Reply:
This is now implemented using symlink directories (and junctions on Windows) here: #13531.